### ISSUE TITLE
Synthetic deinitializer implementation fails on unconstrained generic types

### ISSUE THREAD

**Issue by kyouko-taiga** [hylo-lang/hylo#1262]
This program should compile:

```hylo
trait P { fun bar() }

type A<T>: Deinitializable {
  var x: T
  public memberwise init
}

type B: P, Deinitializable {
  public memberwise init
  public fun bar() {}
}

conformance A: P where T: P {
  fun bar() { x.bar() }
}

public fun main() {
  let a = A<B>(x: B())
  a.bar()
}
```

But the following error is reported during IR analysis:

```
:1.1-19: error: type 'T' does not conform to trait 'Deinitializable'
/* module: test */
~~~~~~~~~~~~~~~~~~
```

**Comment by kyouko-taiga** [comment]
The problem is that the IR is trying to synthesize A<T>.deinitialize, but `T` is not deinitializable.

**Comment by kyouko-taiga** [comment]
Fixed by #1264
